File: LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/[VB]-LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/VB/Vision Clinic/Common/UserCode/Invoice.vb

Namespace LightSwitchApplication

    Public Class Invoice

        Private Sub Invoice_Created()
            InvoiceDate = Date.Today
            InvoiceDue = Date.Today.AddDays(30)
            ShipDate = Date.Today.AddDays(3)
        End Sub

        Private Sub InvoiceDate_Changed()
            InvoiceDue = InvoiceDate.AddDays(30)
            ' If the ShipDate is earlier than the new InvoiceDate, update it.
            If ShipDate < InvoiceDate Then
                ShipDate = InvoiceDate.AddDays(2)
            End If
        End Sub

        Private Sub Tax_Compute(ByRef result As Decimal)
            Result = GetSubTotal() * 0.095
        End Sub
        Protected Function GetSubTotal() As Decimal
            GetSubtotal = 0
            For Each item In InvoiceDetails
                GetSubTotal = GetSubTotal + item.SubTotal
            Next
        End Function

        Private Sub Total_Compute(ByRef result As Decimal)
            ' Set result to the desired field value
            result = GetSubTotal() + Tax
        End Sub
    End Class

End Namespace

File: LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/[C#]-LightSwitch in Visual Studio 2012 Vision Clinic Walkthrough & Sample/C#/Vision Clinic CSharp/Common/UserCode/Invoice.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.LightSwitch;
namespace LightSwitchApplication
{
    public partial class Invoice
    {
        partial void Invoice_Created()
        {
            InvoiceDate = System.DateTime.Today;
            InvoiceDue = System.DateTime.Today.AddDays(30);
            ShipDate = System.DateTime.Today.AddDays(3);
        }

        partial void InvoiceDate_Changed()
        {
            InvoiceDue = InvoiceDate.AddDays(30);
            // If the ShipDate is earlier than the new InvoiceDate, update it
            if (ShipDate < InvoiceDate)
            {
                ShipDate = InvoiceDate.AddDays(2);
            }
        }

        partial void Tax_Compute(ref decimal result)
        {
            result = GetSubTotal() * (decimal)0.095;

        }
        protected decimal GetSubTotal()
        {
            return this.InvoiceDetails.Sum(i => i.SubTotal);
        }

        partial void Total_Compute(ref decimal result)
        {
            result = GetSubTotal() + Tax;
        }   
    }
}
